### ISSUE TITLE
Report warnings

### ISSUE THREAD

**Issue by jyutzler** [opengeospatial/ets-gpkg12#72]
In a few places, we want to report warnings. It is unclear to me how to do this through the interface.
* [ ] [Broken RTree trigger](https://github.com/opengeospatial/geopackage/issues/414)
* ~~If accepted, [require extents for feature tables](https://github.com/opengeospatial/geopackage/pull/450#issuecomment-398850235)~~

**Comment by lgoltz** [comment]
Currently it is only possible to pass, fail or skip a test. 

Warnings can be logged only in the application log.

**Comment by jyutzler** [comment]
I'd like to escalate this. The inability to post a warning has a significant impact in our ability to inform the users of these tests.

**Comment by ghobona** [comment]
@keshav-nangare  @dstenger  There is a log.xml file that is generated by each ETS. Could we use that to log warnings into?

**Comment by dstenger** [comment]
If we write the warning into the log file it is probably hard for non-technical users to find them.

@jyutzler What is the exact purpose of the warnings? Shall they be displayed in the HTML report?

**Comment by jyutzler** [comment]
Yes, I would certainly like for warnings to be displayed in the HTML report. I wouldn't want a GeoPackage flagged with one or more warnings to fail certification, but I also would want responsible organizations to strive to achieve GeoPackages that don't have warnings.

**Comment by dstenger** [comment]
I propose to use a skip result for warnings. A skipped test does not cause an implementation to fail the test suite but a skip is also not a successfully passed test.
The message of the skip might be something like: "WARNING: Some text."

**Comment by dstenger** [comment]
@jyutzler What do you think of this proposal?

**Comment by jyutzler** [comment]
While I do think it would be useful to report on the reason a test was skipped (not requested, no relevant data, etc.), I think this is semantically quite different from a warning. This approach will not allow us to express the message we want to express.

**Comment by dstenger** [comment]
Currently, it is not possible to configure a text when a test passes. Would it help you if we enable that? Then you could configure something like "WARNING: Some text." for your specific case.

Can you please give us an exact example including the actual message?

**Comment by jyutzler** [comment]
RTree:
> WARNING: the trigger `rtree_<t>_<c>_update3` is invalid. Please see http://www.geopackage.org/spec121/#_define_triggers_to_maintain_spatial_index_values for more information.

Once we have proven the concept, we will provide additional instances.

**Comment by dstenger** [comment]
@keshav-nangare Can you please check if such messages (https://github.com/opengeospatial/ets-gpkg12/issues/72#issuecomment-749553658) can be printed by the HTML report of TEAM Engine?
I propose to rename field "Reason of Failure:" to "Details:" and to print the message there. This should be discussed with @ghobona.

**Comment by ghobona** [comment]
@dstenger @keshav-nangare Yes, I am happy for you to rename the field to "Details".

**Comment by dstenger** [comment]
@keshav-nangare 

Can you please check if SoftAssert of TestNG framework can be used?

Steps for implementation:
1. Enhance TEAM Engine to be capable to print warnings/ SoftAssert and rename column "Reason of Failure:" to "Details:".
2. Enhance this test suite to throw a warning (demo).
